plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerLaunchConfigurationDelegate.java

```diff
@@ -90,6 +109,150 @@ private static void validateCloudSdk() throws CoreException  {
   }
 
   @Override
+  public ILaunch getLaunch(ILaunchConfiguration configuration, String mode) throws CoreException {
+    IServer thisServer = ServerUtil.getServer(configuration);
+    DefaultRunConfiguration thisConfig = generateServerRunConfiguration(configuration, thisServer);
+
+    for (ILaunch launch : getLaunchManager().getLaunches()) {
+      if (launch.isTerminated()
+          || launch.getLaunchConfiguration().getType() != configuration.getType()) {
+        continue;
+      }
+      IServer otherServer = ServerUtil.getServer(launch.getLaunchConfiguration());
+      DefaultRunConfiguration otherConfig =
+          generateServerRunConfiguration(launch.getLaunchConfiguration(), otherServer);
+      IStatus conflicts = checkConflicts(thisConfig, otherConfig,
+          new MultiStatus(Activator.PLUGIN_ID, 0,
+              MessageFormat.format("Conflicts with running server \"{0}\"", otherServer.getName()),
+              null));
+      if (!conflicts.isOK()) {
+        throw new CoreException(StatusUtil.filter(conflicts));
+      }
+    }
+    return super.getLaunch(configuration, mode);
+  }
+
+  /**
+   * Create a CloudSdk RunConfiguration corresponding to the launch configuration and server
+   * defaults.
+   */
+  @VisibleForTesting
+  DefaultRunConfiguration generateServerRunConfiguration(ILaunchConfiguration configuration,
+      IServer server) throws CoreException {
+
+    DefaultRunConfiguration devServerRunConfiguration = new DefaultRunConfiguration();
+    // Iterate through our different configurable parameters
+    // TODO: storage-related paths, incl storage_path and the {blob,data,*search*,logs} paths
+
+    // TODO: allow setting host from launch config
+    if (server.getHost() != null) {
+      devServerRunConfiguration.setHost(server.getHost());
+    }
+
+    // TODO: make this a configurable option in the launch config?
+    // default to 1 instance to simplify debugging
+    devServerRunConfiguration.setMaxModuleInstances(1);
+
+    // don't restart server when on-disk changes detected
+    devServerRunConfiguration.setAutomaticRestart(false);
+
+
+    int serverPort = getPortAttribute(LocalAppEngineServerBehaviour.SERVER_PORT_ATTRIBUTE_NAME,
+        LocalAppEngineServerBehaviour.DEFAULT_SERVER_PORT, configuration, server);
+    if (serverPort >= 0) {
+      devServerRunConfiguration.setPort(serverPort);
+    }
+
+    int adminPort =
+        configuration.getAttribute(LocalAppEngineServerBehaviour.ADMIN_PORT_ATTRIBUTE_NAME, -1);
```

Comment:
I believe you forget to refactor this to just call `getPortAttribute()` below? (However, it will handle the exception from `getAttribute()` differently.)